src/diagnostics/MOM_diagnostics.F90

```diff
@@ -951,6 +967,54 @@ subroutine calculate_vertical_integrals(h, tv, p_surf, G, GV, US, CS)
     if (CS%id_col_mass > 0) call post_data(CS%id_col_mass, mass, CS%diag)
   endif
 
+  if (present(Time) .and. GV%Boussinesq) then
+    do j=js,je ; do i=is,ie ; pres(i,j) = 0.0 ; enddo ; enddo
+    do j=js,je ; do i=is,ie ; pres_int(i,j) = 0.0 ; enddo ; enddo
+    call find_eta(h, tv, G, GV, US, eta)
+    call HA_accum('eta', eta, Time, G, HA_CSp)
+    if (associated(tv%eqn_of_state)) then
+      do k=1,nz
+        if (k==1) then
+          do j=js,je ; do i=is,ie
+            z_top(i,j) = eta(i,j)
+            z_bot(i,j) = z_top(i,j) - GV%H_to_Z * h(i,j,k)
+          enddo ; enddo
+        else
+          do j=js,je ; do i=is,ie
+            z_top(i,j) = z_bot(i,j)
+            z_bot(i,j) = z_top(i,j) - GV%H_to_Z * h(i,j,k)
+          enddo ; enddo
+        endif ! (k==1)
+        call int_density_dz(tv%T(:,:,k), tv%S(:,:,k), z_top, z_bot, GV%Rho0, GV%Rho0, &
+                            GV%g_Earth, G%HI, tv%eqn_of_state, US, dpres, dpres_int)
+        do j=js,je ; do i=is,ie
+          pres_int(i,j) = pres_int(i,j) + dpres_int(i,j) + GV%H_to_Z * h(i,j,k) * pres(i,j)
+          pres(i,j) = pres(i,j) + dpres(i,j)
+        enddo; enddo
+      enddo ! k=1,nz
+      do j=js,je ; do i=is,ie
+        bsl(i,j) = pres_int(i,j) / ((GV%g_Earth * GV%Rho0) * (eta(i,j) - z_bot(i,j)))
+      enddo ; enddo
+    else
+      do j=js,je ; do i=is,ie
+        z_top(i,j) = eta(i,j)
+        pres(i,j) = (GV%g_prime(1) * GV%H_to_Z) * (z_top(i,j) + G%bathyT(i,j))
+        pres_int(i,j) = pres_int(i,j) - pres(i,j) * z_top(i,j)
+      enddo ; enddo
+      do k=2,nz ; do j=js,je ; do i=is,ie
+        z_top(i,j) = z_top(i,j) - GV%H_to_Z * h(i,j,k-1)
+        pres(i,j) = (GV%g_prime(k) * GV%H_to_Z) * (z_top(i,j) + G%bathyT(i,j))
+        pres_int(i,j) = pres_int(i,j) - pres(i,j) * z_top(i,j)
+      enddo ; enddo ; enddo
+      do j=js,je ; do i=is,ie
+        bsl(i,j) = pres_int(i,j) / (GV%g_Earth * ((eta(i,j) + G%bathyT(i,j)) + &
```

Comment:
Could you add the equation for BSL in the comment? I trust you must have tested both EoS and layer-mode versions. But it would help me understand the code better with a little bit of context (is the equation $\eta_{BSL} = -\frac{1}{2\rho_0 h}\int^{\rho_b}_{\rho_s} (z+D)^2 d\rho$ ?).